sympy/core/__init__.py

```diff
@@ -33,3 +33,54 @@
 EulerGamma = S.EulerGamma
 GoldenRatio = S.GoldenRatio
 TribonacciConstant = S.TribonacciConstant
+
+__all__ = [
+'sympify', 'SympifyError',
+
+'cacheit',
+
+'Basic', 'Atom', 'preorder_traversal',
+
+'S',
+
+'Expr', 'AtomicExpr', 'UnevaluatedExpr',
+
+'Symbol', 'Wild', 'Dummy', 'symbols', 'var',
+
+'Number', 'Float', 'Rational', 'Integer', 'NumberSymbol',
+'RealNumber', 'igcd', 'ilcm', 'seterr', 'E', 'I', 'nan', 'oo', 'pi', 'zoo',
+'AlgebraicNumber', 'comp', 'mod_inverse',
+
+'Pow', 'integer_nthroot', 'integer_log',
+
+'Mul', 'prod',
+
+'Add',
+
+'Mod',
+
+'Rel', 'Eq', 'Ne', 'Lt', 'Le', 'Gt', 'Ge',
+'Equality', 'GreaterThan', 'LessThan', 'Unequality', 'StrictGreaterThan',
+'StrictLessThan',
+
+'vectorize',
+
+'Lambda', 'WildFunction', 'Derivative', 'diff', 'FunctionClass',
+'Function', 'Subs', 'expand', 'PoleError', 'count_ops',
+'expand_mul', 'expand_log', 'expand_func',
+'expand_trig', 'expand_complex', 'expand_multinomial', 'nfloat',
+'expand_power_base', 'expand_power_exp', 'arity',
+
+'evalf', 'PrecisionExhausted', 'N',
```

Comment:
⚠️ IMHO this should *not* be in the general namespace. It is a huge source of confusion that `N(x, n)` is not a synonym for `evalf(x, n)` it is equivalent to `S(x).evalf(n)`. `evalf` takes *binary* precision and requires options to be given. It should really be a private method. Most other functions follow the `foo(x) ~ x.foo()` idiom. `evalf` does not.

Reply:
I didn't even realize there was a *function* `evalf`, distinct from the module or method. Perhaps we should rename it to `_evalf` or `evalf_binary` and treat it as private API. Is it used anywhere outside of evalf.py? I didn't see anywhere, but it's not easy to grep for.

Reply:
Apparently `evalf` (from `from sympy import *`) in this branch is still the module. I don't know if that was intentional.